functions.photon.php

```diff
@@ -41,11 +41,16 @@ function jetpack_photon_url( $image_url, $args = array(), $scheme = null ) {
 		$args = rawurlencode_deep( $args );
 	}
 
+	/** This filter is documented below. */
+	$custom_photon_url = apply_filters( 'jetpack_photon_domain', '', $image_url );
+
 	// You can't run a Photon URL through Photon again because query strings are stripped.
 	// So if the image is already a Photon URL, append the new arguments to the existing URL.
-	if ( in_array( $image_url_parts['host'], array( 'i0.wp.com', 'i1.wp.com', 'i2.wp.com' ) ) ) {
+	if (
+		$image_url_parts['host'] === parse_url( $custom_photon_url, PHP_URL_HOST )
+		|| in_array( $image_url_parts['host'], array( 'i0.wp.com', 'i1.wp.com', 'i2.wp.com' ) )
```

Comment:
What about flipping the conditional to check for the wp.com first in this same structure? Since it is a OR conditional, if the first condition is met, PHP will keep chugging along without bothering checking the second half of the OR.

Since the wp.com URLs will be used probably 99% of the time, it should save a few cycles here and there.